pkg/operator/ceph/csi/csidriver_object.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	v1k8scsi "k8s.io/api/storage/v1"
+	betav1k8scsi "k8s.io/api/storage/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/storage/v1"
+	"k8s.io/client-go/kubernetes/typed/storage/v1beta1"
+)
+
+type csiDriver interface {
+	createCSIDriverInfo(ctx context.Context, clientset kubernetes.Interface, name, fsGroupPolicy string) error
+	reCreateCSIDriverInfo(ctx context.Context) error
+	deleteCSIDriverInfo(ctx context.Context, clientset kubernetes.Interface, name string) error
+}
+
+type v1CsiDriver struct {
+	csiDriver *v1k8scsi.CSIDriver
+	csiClient v1.CSIDriverInterface
+}
+
+type beta1CsiDriver struct {
+	csiDriver *betav1k8scsi.CSIDriver
+	csiClient v1beta1.CSIDriverInterface
+}
+
+// createCSIDriverInfo Registers CSI driver by creating a CSIDriver object
+func (b1csi beta1CsiDriver) createCSIDriverInfo(ctx context.Context, clientset kubernetes.Interface, name, fsGroupPolicy string) error {
```

Comment:
nit: A single letter is actually convenient in the receiver. How about "d" for "driver"? Both the v1 and v1beta1 types could use that var.
```suggestion
func (d beta1CsiDriver) createCSIDriverInfo(ctx context.Context, clientset kubernetes.Interface, name, fsGroupPolicy string) error {
```

Reply:
Sure, this seems better